tests/e2e/e2e_suite_test.go

```diff
@@ -58,218 +53,285 @@ const (
+				Timeout:       5 * time.Minute,
+				Wait:          true,
+				DryRun:        false,
+			})
+			Expect(err).ShouldNot(HaveOccurred(), "Helm chart installation: %q chart failed with error err: %v", testContext.kubeconfig, err)
 
-	s.T().Log("Test setup teardown completed.")
-}
+			_, _ = fmt.Fprintf(GinkgoWriter, "Helm chart installation: %q completed.\n",
+				testContext.chart)
+			_, _ = fmt.Fprintf(GinkgoWriter, "Helm chart installation: new %q release name.\n",
+				helmReleaseName)
+		})
 
-func (s *Suite) TestDCGMExporter() {
-	s.DCGMExporterPrechecks()
+		labelMap := map[string]string{dcgmExporterPodNameLabel: dcgmExporterPodNameLabelValue}
 
-	s.T().Run("Create workload pod", func(t *testing.T) {
-		t.Log("Creating a workload pod...")
+		It("should create dcgm-exporter pod", func(ctx context.Context) {
+			_, _ = fmt.Fprintln(GinkgoWriter, "Pod creation verification: started")
 
-		var err error
+			Eventually(func(ctx context.Context) bool {
+				pods, err := kubeClient.GetPodsByLabel(ctx, testContext.namespace, labelMap)
+				if err != nil {
+					Fail(fmt.Sprintf("Pod creation: Failed with error: %v", err))
+					return false
+				}
 
-		s.workloadPod, err = s.k8SClient.CreatePod(s.ctx,
-			s.namespace,
-			s.labels,
-			workloadPodName,
-			workloadContainerName,
-			workloadImage,
-		)
+				return len(pods) == 1
+			}).WithPolling(time.Second).Within(15 * time.Minute).WithContext(ctx).Should(BeTrue())
 
-		require.NoError(t, err, "Cannot create workload pod")
+			_, _ = fmt.Fprintln(GinkgoWriter, "Pod creation verification: completed")
+		})
 
-		require.Eventuallyf(t, func() bool {
-			isReady, err := s.k8SClient.CheckPodCondition(s.ctx, s.namespace, s.workloadPod.Name, corev1.PodScheduled)
-			assert.NoErrorf(t, err, "Cannot get pod status: %v", err)
-			return isReady
-		}, 15*time.Minute, 5*time.Second, "Failed to create pod: %s", s.workloadPod.Name)
+		It("should read pod", func(ctx context.Context) {
+			_, _ = fmt.Fprintln(GinkgoWriter, "Read dcgm-exporter pod: started")
+			pods, err := kubeClient.GetPodsByLabel(ctx, testContext.namespace, labelMap)
+			if err != nil {
+				Fail(fmt.Sprintf("Pod creation: Failed with error: %v", err))
+			}
 
-		t.Log("The workload was created.")
-	})
+			Expect(pods).Should(HaveLen(1))
+
+			dcgmExpPod = &pods[0]
+
+			_, _ = fmt.Fprintln(GinkgoWriter, "Read dcgm-exporter pod: completed")
+		})
+
+		It("should ensure that the dcgm-exporter pod is ready", func(ctx context.Context) {
+			_, _ = fmt.Fprintln(GinkgoWriter, "Checking pod status: started")
+			Eventually(func(ctx context.Context) bool {
+				isReady, err := kubeClient.CheckPodStatus(ctx,
+					testContext.namespace,
+					dcgmExpPod.Name,
+					func(namespace, podName string, status corev1.PodStatus) (bool, error) {
+						for _, c := range status.Conditions {
+							if c.Type != corev1.PodReady {
+								continue
+							}
+							if c.Status == corev1.ConditionTrue {
+								return true, nil
+							}
+						}
 
```

Comment:
@rohit-arora-dev, We must wait for the status: PodReady or "CrashLoopBackOff." The check for the  "CrashLoopBackOff" is necessary to interrupt the waiting if the pod crashes. If the pod crashes, waiting until it is up and running doesn't make sense. With the implemented logic, I reduced the waiting time from 15 minutes to a few seconds in case we see that the pod crashes.